components/gitpod-cli/cmd/tasks.go

```
@@ -0,0 +1,172 @@
+	"github.com/spf13/cobra"
+	"golang.org/x/term"
+	"google.golang.org/grpc"
+)
+
+// tasksCmd represents the tasks command
+var tasksCmd = &cobra.Command{
+	Use:   "tasks",
+	Short: "Interact with workspace tasks",
+	Run: func(cmd *cobra.Command, args []string) {
+		if len(args) == 0 {
+			cmd.Help()
+		}
+	},
+}
+
+var attachTaskCmdOpts struct {
+	Interactive bool
+	ForceResize bool
+}
+
+// TODO(andreafalzetti): refactor tasks.go to only keep the tasks cmd initialisation, moving the subcommands in their own file
+
+// This func was copied from https://github.com/gitpod-io/gitpod/blob/main/components/supervisor/cmd/terminal.go#L29
+// TODO(andreafalzetti): move it somewhere else so that it's reusable from other cmds
+func dialSupervisor() *grpc.ClientConn {
+	// TODO(andreafalzetti): figure out how to import peroperly supervisor.GetConfig()
+	// cfg, err := supervisor.GetConfig()
+	// if err != nil {
+	// 	log.WithError(err).Fatal("cannot get config")
+	// }
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	url := fmt.Sprintf("localhost:%d", 22999)
+	conn, err := grpc.DialContext(ctx, url, grpc.WithInsecure(), grpc.WithBlock())
+	if err != nil {
+		log.WithError(err).Fatal("cannot connect to supervisor")
+	}
+
+	// TODO(cw): devise some means to properly close the connection
+
+	return conn
+}
+
+// listTasksCmd represents the tasks list command
+var listTasksCmd = &cobra.Command{
+	Use:   "list",
+	Short: "Lists the workspace tasks and their status",
+	Run: func(cmd *cobra.Command, args []string) {
+		var (
+			client = api.NewStatusServiceClient(dialSupervisor())
+		)
+
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
+
+		// TODO(andreafalzetti): ask how to opt-out from the stream! {Observe: false} gives me a stream 😢
+		listen, err := client.TasksStatus(ctx, &api.TasksStatusRequest{Observe: false})
+		if err != nil {
+			fmt.Println("Cannot list tasks")
```

we shoudl not swallow errors, but print them as well, looks how other CLI commands do it

---

fixed in https://github.com/gitpod-io/gitpod/pull/7601/commits/0fc8c8dfc6560cd8ab4c9379ba1370147740280f